### PR TITLE
Coe 933 oee simulators update instruction readme md on how to configure simulator definition and shiftplan

### PR DESCRIPTION
**Pull request by namnhatpham**

### CODE REVIEW

File: simulators/README.md [hunk 1]
```diff
@@ -9,14 +9,54 @@ There are extra features in [extras](extras):
 - the **import data** to upload Measurements or/and Alarms from data json file to OEE profiles.
 
 ## Simulator Microservice
+Detailed feature list:
+- configuration in JSON, no need to write code.
+- automatically creates devices and sends data.
+- identifies devices using a configurable `externalId`.
+- devices can be disabled to not send any events and measurements.
+- Written in python which can be modified easily for further development.
+- Simulators work based on status and shift that it is assigned to.
```
Comment (mvoigt-sag) [inline simulators/README.md:18]: please mention that the shift is linked via the locationId

File: simulators/README.md [hunk 2]
```diff
@@ -52,24 +100,57 @@ Detailed feature list:
...
+      ```
+      "measurements": [
+          "type": "PumpPressure",
+          "fragment": "Pressure",
+          "series": "P"
+      ]
+      ```
+    where "type" is optional and its default value is the value from the "fragment" property.
+  
+  - In measurements, `valueDistribution` is defined to let the simulator know which distribution formular to use to generate measurements. There are three choices that can be defined here: `uniform`, `uniformint`, `normal`.
```
Comment (mvoigt-sag) [inline simulators/README.md:113]: "formular" = "formula" ?

File: simulators/README.md [hunk 3]
```diff
@@ -52,24 +100,57 @@ Detailed feature list:
...
+  
 - Simulators act according to given Shiftplans
   - Simulators are linked to shiftplans via locationId
-  - Shiftplans are polled once a day
-  - If a Simulator is not in Production time according to the given Shiftplan, it will not produce any  events
-  - At startup [shiftplans.json](main/shiftplans.json) is parsed and used to update the given timeslots whitin the shiftplan
-  - The `locationId's` presented in [shiftplans.json](main/shiftplans.json) are used to parse for all shiftplans used in the script and are the only shiftplans are considered for the polling
+  - If a Simulator is not in Production time according to the given Shiftplan, it will not produce any events and measurements.
+  - At startup [shiftplans.json](main/shiftplans.json) is parsed and used to update the given timeslots within the shiftplan.
+  - The `locationId's` presented in [shiftplans.json](main/shiftplans.json) are used to parse for all shiftplans used in the script but only a specific shiftplan is applied to a simulator if it define the `locationId` of a shift.
```
Comment (mvoigt-sag) [inline simulators/README.md:125]: I think we need to change that a little bit:
- we need to mention that the shiftplans.json is parsed once after startup and the shifts are created accordingly (if they dont exist yet)
- everytime an event/measurement is about to be sent, the script checks if the status of the locationId == PRODUCTION; and only if the status is correct, the event/measurement is sent
- so all changes to the shiftplan are taken into account

File: simulators/README.md [hunk 4]
```diff
@@ -52,24 +100,57 @@ Detailed feature list:
...
+      "recurringTimeslots": [
+        { "id": "OneShiftLocation-DayShift", "seriesPostfix": "DayShift", "slotType": "PRODUCTION", "slotStart": "2022-07-13T08:00:00Z", "slotEnd": "2022-07-13T16:00:00Z", "description": "Day Shift", "active": true, "slotRecurrence": { "weekdays": [1, 2, 3, 4, 5] } },
+        { "id": "OneShiftLocation-Break", "slotType": "BREAK", "slotStart": "2022-07-13T12:00:00Z", "slotEnd": "2022-07-13T12:30:00Z", "description": "Day Shift Break", "active": true, "slotRecurrence": { "weekdays": [1, 2, 3, 4, 5] } }
+      ]
+    },
+    {...}
+  ]
+```
+This is an example of a shiftplan which contains a shift name `OneShiftLocation` in `locationId` field.
+The `recurringTimeslots` field is an array which define components of the shift. In this case, this shift has two parts:
```
Comment (mvoigt-sag) [inline simulators/README.md:151]: I guess it is worth mentioning that only "recurringTimeslots" is supported; "timeslots" is not supported (pls verify this statement in the code)

**Review by mvoigt-sag — CHANGES_REQUESTED** [review]

**Comment by mvoigt-sag** [comment]
@MarkJMReynolds-SAG maybe you could have a look at this PR and also https://github.com/SoftwareAG/oee-simulators/blob/Article/Article.md ? Thanks!

File: simulators/README.md (hunk 1 above)
Comment (namnhatpham) [reply simulators/README.md:18]: It is already mentioned  in the explanation of simulator
```suggestion
- Simulators act according to given Shiftplans
  - Simulators are linked to shiftplans via locationId
```

**Review by mvoigt-sag — APPROVED** [review]

**Pull request merged**